File: core/src/main/java/com/animation/morphtransitions/FloatProperty.java
package com.animation.morphtransitions;

import android.util.Property;

abstract class FloatProperty<T> extends Property<T, Float> {
    public FloatProperty(String name) {
        super(Float.class, name);
    }

    /**
     * A type-specific override of the {@link #set(Object, Float)} that is faster when dealing
     * with fields of type <code>float</code>.
     */
    public abstract void setValue(T object, float value);

    @Override
    final public void set(T object, Float value) {
        setValue(object, value);
    }
}

File: core/src/main/java/com/animation/morphtransitions/FloatProperty.kt
package com.animation.morphtransitions

import android.util.Property

internal abstract class FloatProperty<T>(name: String?)
    : Property<T, Float>(Float::class.java, name) {
    /**
     * A type-specific override of the [.set] that is faster when dealing
     * with fields of type `float`.
     */
    abstract fun setValue(obj: T, value: Float)

    override fun set(obj: T, value: Float) {
        setValue(obj, value)
    }
}
